libs/designer/src/lib/core/actions/bjsworkflow/add.ts

```diff
@@ -332,11 +329,16 @@ const getOperationType = (operation: DiscoveryOperation<DiscoveryResultTypes>):
     : operationType;
 };
 
-export const getRootNodeDetails = (workflowState: WorkflowState, connectorId: string): string | undefined => {
-  if (workflowState.graph?.children && connectorId.indexOf(Constants.INVOKER_CONNECTION.DATAVERSE_CONNECTOR_ID) > -1) {
+export const getRootNodeManifest = async (
+  workflowState: WorkflowState,
+  operations: OperationMetadataState
+): Promise<OperationManifest | undefined> => {
+  if (workflowState?.graph?.children !== undefined) {
     for (const child of workflowState.graph.children) {
       if (workflowState.graph?.id === 'root') {
-        return child.id;
+        const rootNodeOperationInfo = operations.operationInfo[child.id];
+        const { connectorId, operationId } = rootNodeOperationInfo;
+        return await getOperationManifest({ connectorId, operationId });
       }
     }
```

Comment:
Update code to
`const triggerNodeId = getTriggerNodeId(workflowState);
 const operationInfo = operations.operationInfo[triggerNodeId]
 if (operationInfo && OperationManifestService().isSupported(operationInfo.type, operationInfo.kind)) {
     return getOperationManifest(operationInfo.connectorId, operationInfo.operationId);
 }
 return undefined;
 `